### PR TITLE
Formplayer To Master Merge

### PR DESCRIPTION
**Pull request by shubham1g5**
Brings `master` up to date with `formplayer`. 

Additional Commits on top of `formplayer`:

- https://github.com/dimagi/commcare-core/pull/1120/commits/9077335df7bdde781fb3ec34557589c4b6159ce3

cross-request: https://github.com/dimagi/commcare-android/pull/2558

### CODE REVIEW

**Comment by snopoke** [comment]
What's the strategy for reviewing this?

**Comment by snopoke** [comment]
I haven't reviewed this yet but before I do can you confirm how you created this branch? From the description it seems like you may have branched off `formplayer` and then added 9077335 on top. Is that correct?

**Comment by shubham1g5** [comment]
@snopoke Kind of, I branched off `master` and then merged `formplayer` in it resolving conflicts. I then added https://github.com/dimagi/commcare-core/commit/9077335df7bdde781fb3ec34557589c4b6159ce3 followed by more merges from `master` and `formplayer` to keep things up to date on this branch.

File: src/main/java/org/javarosa/core/model/FormDef.java [hunk 1]
```diff
@@ -1101,9 +1098,14 @@ private void evaluateTriggerables(List<Triggerable> tv,
      * @param triggerable The triggerable to be updated
      * @param anchorRef   The reference to the value which was changed.
      */
+    @Trace
```
Comment (snopoke) [inline src/main/java/org/javarosa/core/model/FormDef.java:1101]: are these trace annotations going to be OK on mobile?

**Review by snopoke — COMMENTED** [review]
Phew 😅! A few small comments but nothing major.

Given the serialization changes what will the impact on existing installations be of rolling this out?

File: src/main/java/org/javarosa/core/model/FormDef.java (hunk 1 above)
Comment (shubham1g5) [reply src/main/java/org/javarosa/core/model/FormDef.java:1101]: I believe so - https://github.com/dimagi/commcare-core/pull/1063#issuecomment-1011491637 , I have not faced any issues so far while testing mobile with this branch.

**Comment by shubham1g5** [comment]
Thanks @snopoke, it seems like you gave the whole thing a read which sounds quite tiresome. I am going to make a PR into `formplayer` itself to address the nits and will do another merge on `formplayerUpdate` from `formplayer` to get some last set of changes in that has happened recently on `formplayer`

**Comment by shubham1g5** [comment]
> Given the serialization changes what will the impact on existing installations be of rolling this out?

It's okay since all the serialization changes are around Suite.xml models and Mobile builds suite and profile config xml models from xml on every run without storing the serialised object in DB.

File: src/main/java/org/commcare/core/process/CommCareInstanceInitializer.java [hunk 2]
```diff
@@ -187,8 +260,8 @@ public String getVersionString() {
         return "CommCare Version: " + mPlatform.getMajorVersion() + "." + mPlatform.getMinorVersion();
     }
 
-    protected AbstractTreeElement setupMigrationData(ExternalDataInstance instance) {
-        return null;
+    protected InstanceRoot setupMigrationData(ExternalDataInstance instance) {
```
[the remark this answers is not in the document]
Comment (shubham1g5) [reply src/main/java/org/commcare/core/process/CommCareInstanceInitializer.java:263]: i can't remember what was this used for, just double checking if we are sure it's ok to remove this code ?

[the remark this answers is not in the document]
Comment (snopoke) [reply src/main/java/org/commcare/core/process/CommCareInstanceInitializer.java:263]: This was part of initial work for bulk data migrations that never happened.

**Review by snopoke — DISMISSED** [review]

**Review by snopoke — APPROVED** [review]

**Pull request merged**